### PR TITLE
Avoid destination's file outside repository

### PR DESCRIPTION
**Pull request by Dioxo**
## :loudspeaker: Type of change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## :scroll: Description

As I said before, user can move a password outside the repository by using `../` This should fix it.

## :pencil: Checklist

- [x] I formatted the code with the IDE's reformat action (Ctrl + Shift + L/Cmd + Shift + L)
- [x] I reviewed submitted code
- [ ] I added a [CHANGELOG](CHANGELOG.md) entry if applicable

### CODE REVIEW

File: app/src/main/java/com/zeapo/pwdstore/crypto/PasswordCreationActivity.kt [hunk 1]
```diff
@@ -235,6 +236,11 @@ class PasswordCreationActivity : BasePgpActivity(), OpenPgpServiceConnection.OnB
                                 snackbar(message = getString(R.string.password_creation_duplicate_error))
                                 return@executeApiAsync
                             }
+
+                            if (!URI(path).normalize().path.contains(repoPath)) {
+                                snackbar(message = getString(R.string.message_error_destination_outside_repo))
+                                return@executeApiAsync
+                            }
```
Comment (msfjarvis) [inline app/src/main/java/com/zeapo/pwdstore/crypto/PasswordCreationActivity.kt:243]: why not use this from the previous PR? Declare it as an extension function and re-use it.

```kotlin
private fun isInsideRepository(file: File): Boolean {
    return file.canonicalPath.contains(getRepositoryDirectory(this).canonicalPath)
}
```

Comment (Dioxo) [reply app/src/main/java/com/zeapo/pwdstore/crypto/PasswordCreationActivity.kt:243]: I don't really know how to create an extension function... jeje... But I'll search it and change it.

Comment (msfjarvis) [reply app/src/main/java/com/zeapo/pwdstore/crypto/PasswordCreationActivity.kt:243]: Look at the `Extensions.kt` file in the repo, it's pretty straightforward.

Comment (Dioxo) [reply app/src/main/java/com/zeapo/pwdstore/crypto/PasswordCreationActivity.kt]: I think I did it :), if you can please check.

File: app/src/main/java/com/zeapo/pwdstore/utils/Extensions.kt [hunk 2]
```diff
@@ -123,3 +124,7 @@ fun <T : View> AlertDialog.requestInputFocusOnView(@IdRes id: Int) {
 val Context.autofillManager: AutofillManager?
     @RequiresApi(Build.VERSION_CODES.O)
     get() = getSystemService()
+
+fun Activity.isInsideRepository(file: File): Boolean {
+    return file.canonicalPath.contains(getRepositoryDirectory(this).canonicalPath)
+}
```
Comment (msfjarvis) [inline app/src/main/java/com/zeapo/pwdstore/utils/Extensions.kt:130]: needs a trailing newline

Comment (Dioxo) [reply app/src/main/java/com/zeapo/pwdstore/utils/Extensions.kt:130]: You mean I should put another line above the function?

Comment (msfjarvis) [reply app/src/main/java/com/zeapo/pwdstore/utils/Extensions.kt:130]: No, at the end of the file.

**Pull request merged**